### PR TITLE
Make trailing spaces count as errors

### PR DESCRIPTION
**Pull request by udayvir-singh**
fixes the problem when you type space at the end and it doesn't count as error until you have typed the next word, which fells very buggy.

For Example:
```
*-------------------*
* ll hi ll hi ll hi *
*-------------------*

ll_h_
```
won't count as error, until this much is typed:
```
ll_h_i
```

### CODE REVIEW

**Comment by ranelpadon** [comment]
@udayvir-singh Thanks for your PR. Although I think this was similar to my original code:
https://github.com/ranelpadon/ngram-type/commit/264320652370041907a0a44f88b04f3f582f6c74

which is a fix to the trailing space's filed issue:
> There is a delay between typing the last character and advancing to the next test. If you type a space during this time (which is kinda natural) it causes an error instead and the test also fails to advance.
https://github.com/ranelpadon/ngram-type/issues/12

I could still accept your PR, but I'm thinking maybe we could also have a toggle-able option like `Ignore Trailing Space` which is enabled by default like the use case for #12. Then, you could set it to disabled (i.e. trailing space should be counted as error), for your use case. Wdyt?

**Comment by udayvir-singh** [comment]
@ranelpadon
Thanks for reply, just added another commit that fixes problems with #12.

**Comment by ranelpadon** [comment]
@udayvir-singh Sorry, I was quite busy recently, and just realized that your MR is fixing a bug while the phrase is not fully typed yet (mid part), unlike the #12 which is when the phrase is already typed (end part). I thought both of them were on the end parts that's why I was confused. They're indeed different scenarios, and both should work. Can you please open your MR again? Thanks

File: index.html [hunk 1]
```diff
@@ -786,9 +786,9 @@ <h3 class="title-wrapper center card-header"><a href="https://github.com/ranelpa
                         return;
                     }
 
-                    // Remove spaces at both ends of the phrase
+                    // Remove spaces at starting of the phrase
                     // which might be accidentally typed.
-                    var typedPhrase = this.typedPhrase.trim();
+                    var typedPhrase = this.typedPhrase.replace(/^ +/, "");
```
Comment (ranelpadon) [inline index.html:791]: Please use `.trimStart()` which is simpler/more readable:
`var typedPhrase = this.typedPhrase.trimStart()`

File: index.html [hunk 2]
```diff
@@ -804,7 +804,7 @@ <h3 class="title-wrapper center card-header"><a href="https://github.com/ranelpa
                         this.isInputCorrect = true;
                         this.hitsCorrect += 1;
                     }
-                    else {
+                    else if (this.expectedPhrase !== typedPhrase.replace(/ +$/, "")) {
```
Comment (ranelpadon) [inline index.html:807]: Please use `.trimEnd()` which is simpler/more readable:
`else if (this.expectedPhrase !== typedPhrase.trimEnd()) {`

File: index.html [hunk 3]
```diff
@@ -814,10 +814,7 @@ <h3 class="title-wrapper center card-header"><a href="https://github.com/ranelpa
                         this.hitsWrong += 1;
                     }
 
-                    if (
-                        typedPhrase.length === this.expectedPhrase.length
-                        && typedPhrase === this.expectedPhrase
-                    ) {
+                    if (typedPhrase.replace(/ +$/, "") === this.expectedPhrase) {
```
Comment (ranelpadon) [inline index.html:817]: Please use `.trimEnd()` which is simpler/more readable:
`if (typedPhrase.trimEnd() === this.expectedPhrase) {`

**Review by ranelpadon — CHANGES_REQUESTED** [review]
Done reviewing, changes look good. Just added some minor comments. Will merge once you update it. Thanks :)

**Comment by udayvir-singh** [comment]
@ranelpadon I had deleted my fork so I can't make changes to this pull request.

I have opened another pull request from my new fork see #18.

**Comment by ranelpadon** [comment]
Ah I see, no worries. Will check the other MR. Thanks!

**Pull request closed without merging**